src/site/content/en/blog/racer/index.md

```diff
@@ -0,0 +1,289 @@
+
+```css
+#sprite {
+    height: 100px; 
+    width: 100px;
+    background: url(sprite.jpg) 0 0 no-repeat;
+    -webkit-animation: play-sprite 0.33s linear 0s steps(9) infinite;
+}
+
+@-webkit-keyframes play-sprite {
+    0%     { background-position: 0px 0px; }
+    100%   { background-position: -900px 0px; }
+}
+```
+
+The problem with this technique is you can only use sprite sheets laid out on a single row. In order to loop through multiple rows, the animation must be chained through multiple keyframe declarations.
+
+```css
+#sprite {
+    height: 100px; 
+    width: 100px;
+    background: url(sprite.jpg) 0 0 no-repeat;
+    -webkit-animation-name: row1, row2, row3;
+    -webkit-animation-duration: 0.2s;
+    -webkit-animation-delay: 0s, 0.2s, 0.4s;
+    -webkit-animation-timing-function: steps(5), steps(5), steps(5);
+    -webkit-animation-fill-mode: forwards;
+}
+
+@-webkit-keyframes row1 {
+    0%     { background-position: 0px 0px; }
+    100%   { background-position: -500px 0px; }
+}
+
+@-webkit-keyframes row2 {
+    0%     { background-position: 0px -100px; }
+    100%   { background-position: -500px -100px; }
+}
+
+@-webkit-keyframes row3 {
+    0%     { background-position: 0px -200px; }
+    100%   { background-position: -500px -200px; }
+}
+```
+
+## Rendering the Cars
+
+As with any car racing game we knew it was important to give the user a feeling of acceleration and handling. Applying a different amount of traction was important for game balancing and the fun factor, so that once a player got a feel for the physics, they’d get a sense of accomplishment and become a better racer. 
+
+Once again we called on Paper.js which comes with an extensive set of math utilities. We used some of its methods to move the car along the path, while adjusting the car position and rotation smoothly each frame.
+
+```js
+var trackOffset = _path.length - (_elapsed % _path.length);
+var trackPoint = _path.getPointAt(trackOffset);
+var trackAngle = _path.getTangentAt(trackOffset).angle;
+
+_velocity.length += _throttle; //apply the throttle
+
+if (!_throttle) {
+    //slow down since the throttle is off
+    _velocity.length *= FRICTION;
+}
+
+if (_velocity.length > MAXVELOCITY) {
+    _velocity.length = MAXVELOCITY;
+}
+
+_velocity.angle = trackAngle;
+
+trackOffset -= _velocity.length;
+_elapsed += _velocity.length;
+
+//find if a lap has been completed
+if (trackOffset < 0) {
+    while (trackOffset < 0) trackOffset += _path.length;
+    trackPoint = _path.getPointAt(trackOffset);
+    console.log('LAP COMPLETE!');
+}
+
+if (_velocity.length > 0.1) {
+    //render the car if there is actually velocity
+    renderCar(trackPoint);
+}
```

Comment:
```suggestion
var trackOffset = _path.length - (_elapsed % _path.length);
var trackPoint = _path.getPointAt(trackOffset);
var trackAngle = _path.getTangentAt(trackOffset).angle;

// Apply the throttle
_velocity.length += _throttle;

if (!_throttle) {
  // Slow down since the throttle is off
  _velocity.length *= FRICTION;
}

if (_velocity.length > MAXVELOCITY) {
  _velocity.length = MAXVELOCITY;
}

_velocity.angle = trackAngle;
trackOffset -= _velocity.length;
_elapsed += _velocity.length;

// Find if a lap has been completed
if (trackOffset < 0) {
  while (trackOffset < 0) trackOffset += _path.length;

  trackPoint = _path.getPointAt(trackOffset);
  console.log('LAP COMPLETE!');
}

if (_velocity.length > 0.1) {
  // Render the car if there is actually velocity
  renderCar(trackPoint);
}
```